ci/min_deps_check.py

```diff
@@ -141,21 +141,20 @@ def process_pkg(
     policy_months = POLICY_MONTHS.get(pkg, POLICY_MONTHS_DEFAULT)
     policy_published = datetime.now() - timedelta(days=policy_months * 30)
 
-    policy_major = req_major
-    policy_minor = req_minor
-    policy_published_actual = req_published
-    for (major, minor), published in reversed(sorted(versions.items())):
-        if published < policy_published:
-            break
```

Comment:
I'm not *entirely* sure why this was broken, but the new logic is significantly simpler so I'm more confident it's correct. In particular, it determines the policy versions without using the currently required versions as defaults.

Reply:
An alternative implement of the original policy that is possibly clearer:
```suggestion
    # "the minor version (X.Y) initially published no more than N months ago"
    policy_major, policy_minor = min(
        (version for version, date in versions.items() if date > policy_published),
        default=(req_major, req_minor),
    )
```